### ISSUE TITLE
Config class should not modify names of properties

### ISSUE THREAD

**Issue by oskarwrobel** [ckeditor/ckeditor5-utils#43]
Right now every property added to the config is transformed to lowercase.
```javascript
config.set( {
    resize: {
        minHeight: 400
    }
} );

expect( config.get( 'resize' ) ).to.be.deep.equal( {
    minheight: 400
});
```

It could make some problems:
```
config.set( {
  data: {
    fooBar: 'biz'
  }
} );

const data = config.get( 'data' );
data.fooBar // undefined
```

Transformation to lower case should be done only for comparison inside `get()` method. Or maybe  we don't need this transformation at all?

Noticed by @Reinmar here: https://github.com/ckeditor/ckeditor5-utils/pull/36#issuecomment-225824494

**Comment by pomek** [comment]
> Transformation to lower case should be done only for comparison inside get() method. Or maybe we don't need this transformation at all?

So, what is the decision?

**Comment by Reinmar** [comment]
Since we have a clear naming scheme in CKEditor 5 (CKEditor 4's was super confusing), I guess we could totally remove the normalisation.

But I've just started thinking for whom the normalisation was added in the first place, and it's clear that for developers who integrate CKEditor with their websites. So the normalisation is needed on `set()` not `get()` what means that this ticket is incorrect...

However, if we keep normalising on `set()` we'll introduce situations like the one in the tests where magically object properties get changed. This is unacceptable. Imagine that you use CKE's config object to store some bigger object that your feature pass to e.g. an external library. That's going to break super fast.

So, to sum up – normalisation must be removed as it will break things.

This change can be done under this issue as it still matches the scheme.
